### PR TITLE
fetched informtion from api

### PR DESCRIPTION
**Pull request by m-hashimrana**
fetched user data from api
implemented componentDidMount()
displayed usersList in tabular form (id,name, username,email,phone,website)
code refactored

### CODE REVIEW

File: src/components/User.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import React, { useState } from 'react';
+
+const User = ({ data }) => {
+	return (
+		<div>
+			<table className='table'>
+				<thead className='thead-light'>
+					<tr>
+						<th scope='col'>ID</th>
```
Comment (usman-cogent-labs) [inline src/components/User.js:9]: make a reusable component for this

File: src/components/User.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+						<th scope='col'>Phone</th>
+						<th scope='col'>website</th>
+						<th scope='col'>UserName</th>
+					</tr>
+				</thead>
+				<tbody>
+					{data?.map((user) => (
+						<tr key={user.id}>
+							<th scope='row'>{user?.id}</th>
+							<td>{user?.name}</td>
```
Comment (usman-cogent-labs) [inline src/components/User.js:21]: make a reusable component for this too

File: src/components/Users.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import React, { useEffect, useState } from 'react';
+import User from './User';
+
+const Users = () => {
+	const [employees, setEmployees] = useState([]);
+	const [error, setError] = useState(null);
+
+	const fetchUsers = async () => {
+		setError(null);
```
Comment (usman-cogent-labs) [inline src/components/Users.js:9]: remove this line of code

File: src/components/Users.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+		try {
+			const response = await fetch('https://jsonplaceholder.typicode.com/users');
+
+			if (!response.ok) {
```
Comment (usman-cogent-labs) [inline src/components/Users.js:13]: remove this code as well

File: src/components/Users.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+				throw new Error('Something went wrong!');
+			}
+			const data = await response.json();
```
Comment (usman-cogent-labs) [inline src/components/Users.js:16]: check if the data is in json already if yes then please remove this line

File: src/components/Users.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+			const loadedEmplyees = [];
+			for (let keys in data) {
```
Comment (usman-cogent-labs) [inline src/components/Users.js:19]: key*
users*

File: src/components/Users.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+				loadedEmplyees.push({
```
Comment (usman-cogent-labs) [inline src/components/Users.js:20]: remove this formatting and set the data as it is

File: src/components/Users.js [hunk 8]
```diff
@@ -0,0 +1,41 @@
+import React, { useEffect, useState } from 'react';
+import User from './User';
+
+const Users = () => {
+	const [employees, setEmployees] = useState([]);
```
Comment (usman-cogent-labs) [inline src/components/Users.js:5]: change the state name to users

**Review by usman-cogent-labs — CHANGES_REQUESTED** [review]

File: src/components/Users.js (hunk 4 above)
Comment (m-hashimrana) [reply src/components/Users.js]: can't be removed because of fetch limitations

**Review by usman-cogent-labs — APPROVED** [review]

**Pull request merged**